tests/snuba/search/test_backend.py

```diff
@@ -1217,6 +1229,297 @@ def test_first_release_environments(self):
         )
         assert set(results) == set([self.group1])
 
+    def test_first_release_any_or_no_environments(self):
+        # test scenarios for tickets:
+        # SEN-571
+        # ISSUE-432
+
+        # given the following setup:
+        #
+        # groups table:
+        # group    first_release
+        # A        1
+        # B        1
+        # C        2
+        #
+        # groupenvironments table:
+        # group    environment    first_release
+        # A        staging        1
+        # A        production     2
+        #
+        # when querying by first release, the appropriate set of groups should be displayed:
+        #
+        #     first_release: 1
+        #         env=[]: A, B
+        #         env=[production, staging]: A
+        #         env=[staging]: A
+        #         env=[production]: nothing
+        #
+        #     first_release: 2
+        #         env=[]: A, C
+        #         env=[production, staging]: A
+        #         env=[staging]: nothing
+        #         env=[production]: A
+
+        # create some releases
+
+        release_1_timestamp = self.base_datetime
+        release_1 = self.create_release(self.project, date_added=release_1_timestamp)
+
+        release_2_timestamp = release_1_timestamp + timedelta(days=2)
+        release_2 = self.create_release(self.project, date_added=release_2_timestamp)
+
+        assert release_2_timestamp > release_1_timestamp  # release_2 occurred after release_1
+
+        # create an issue/group whose events that occur in 2 distinct environments
+
+        group_a_event_1 = self.store_event(
+            data={
```

Comment:
Just in terms of getting the size of this test down, you can remove the `tags` and `stacktrace` sections from all events you're creating, they're not necessary in this case.